src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -310,6 +313,94 @@ public void terminatesMonitorsAndProcessIfClosedNormal() throws Exception {
     }
 
     /**
+     * VerboseProcess can run a java process which will throw a stack trace and
+     * makes sure it will group the stack trace.
+     */
+    @Test
+    public void logCompleteStackTrace() {
+        final org.apache.log4j.Logger logger =
+                org.apache.log4j.Logger.getRootLogger();
+        final TestAppender appender = new TestAppender();
+        logger.addAppender(appender);
+        final String[] commands = new String[] {
+            retrieveJavaExecLocation(), "-cp",
+            System.getProperty("java.class.path"),
+            "com.jcabi.log.VerboseProcessExample",
+        };
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        final VerboseProcess process = new VerboseProcess(
+            builder, Level.INFO, Level.SEVERE
+        );
+        IllegalArgumentException caught = new IllegalArgumentException();
+        boolean failed = false;
+        try {
+            process.stdout();
+        } catch (final IllegalArgumentException ex) {
+            caught = ex;
+            failed = true;
+        } finally {
+            logger.removeAppender(appender);
+            process.close();
+        }
+        Assert.assertTrue("Process should have failed!", failed);
+        MatcherAssert.assertThat(
+                caught.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_1)
+        );
+        MatcherAssert.assertThat(
+                caught.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_2)
+        );
+        verifyLogs(appender);
+    }
+
+    /**
+     * Gets the location of Java, whether on Linux of Windows.
+     * @return String with Java location
+     */
+    private static String retrieveJavaExecLocation() {
+        final String rootpath = System.getProperty("java.home");
+        if (SystemUtils.IS_OS_WINDOWS) {
+            final String winpath =
+                    rootpath.replaceAll("\\\\", "\\\\\\\\");
+            final String finalpath = String.format(
+                "%s%s", winpath, "\\bin\\java.exe"
+            );
+            final File file = new File(finalpath);
+            if (file.exists()) {
+                return finalpath;
+            }
+        } else {
+            final String linuxpath = String.format(
+                "%s%s", rootpath, "/bin/java"
+            );
+            final File file = new File(linuxpath);
+            if (file.exists()) {
```

Comment:
@dean-e-clark this is code duplication with windows part, we need to unify this code